### PR TITLE
Use a default of `Any` for the `ImportString` type variable

### PR DESCRIPTION
**Pull request by Viicos**
The runtime behavior is to allow anything if `ImportString` isn't parametrized. We reflect this by providing a default value to the type variable, so that type checkers don't complain on missing parametrization.

Requires:
- https://github.com/pydantic/pydantic/issues/10197

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by Viicos** [comment]
I'll have to take a deeper look on how this plays with mypy..

**Review by sydney-runkle — APPROVED** [review]
The king of typing strikes again! Great work.

**Comment by sydney-runkle** [comment]
LGTM once we resolve `mypy` test issues!

**Comment by sydney-runkle** [comment]
Might want to ping DM if you're having trouble with `mypy` stuff.

**Comment by sydney-runkle** [comment]
Closing for now as we're blocked on the `mypy` front. Happy to revisit later :).

**Pull request closed without merging**
